File: app/src/main/java/com/asura/android_study/receiver/NetWorkReceiver.java
package com.asura.android_study.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.asura.a_log.ALog;

/**
 * Created by Liuxd on 2016/9/6 10:04.
 * 网络广播监听者
 */
public class NetWorkReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
            ConnectivityManager mConnectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            NetworkInfo netInfo = mConnectivityManager.getActiveNetworkInfo();
            if (netInfo != null && netInfo.isConnected()) {
                ALog.INSTANCE.e("onReceive", "网络可用" + netInfo.getType() + "    ,getTypeName==" + netInfo.getTypeName()
                        + "getSubtype==" + netInfo.getSubtype());
                //网络可用时同步数据
                //                if (netInfo.getType() == ConnectivityManager.TYPE_WIFI) {
                //                    //WiFi网络
                //                } else if (netInfo.getType() == ConnectivityManager.TYPE_ETHERNET) {
                //                    //有线网络
                //                } else if (netInfo.getType() == ConnectivityManager.TYPE_MOBILE) {
                //                    //3g网络
                //                } else if (netInfo.getSubtype() == TelephonyManager.NETWORK_TYPE_LTE) {
                //                    //4G网络
                //                }
            } else {
                //网络断开
                ALog.INSTANCE.e("onReceive", "网络断开");
            }
        }
    }

    ;

}

File: app/src/main/java/com/asura/android_study/activity/eventbus/NetWorkReceiver.kt
package com.asura.android_study.activity.eventbus

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo

import com.asura.a_log.ALog
import com.asura.android_study.activity.eventbus.event.NetStateEvent
import org.greenrobot.eventbus.EventBus

/**
 * Created by Liuxd on 2018/7/24 18:04.
 * 网络广播监听者
 */
//class NetWorkReceiver(private val mCallback: INetworkStateChangeCallback) : BroadcastReceiver() {
class NetWorkReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action != null && intent.action == ConnectivityManager.CONNECTIVITY_ACTION) {
            val mConnectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            var netInfo: NetworkInfo? = null
            if (mConnectivityManager != null) {
                netInfo = mConnectivityManager.activeNetworkInfo
            }
            if (netInfo != null && netInfo.isConnected) {
                ALog.d("onReceive :网络可用")
//                mCallback.onNetworkAvailable(true)
                EventBus.getDefault().post(NetStateEvent(NetState.NET_AVAILABLE))
            } else {
                ALog.d("onReceive :网络断开")
//                mCallback.onNetworkAvailable(false)
                EventBus.getDefault().post(NetStateEvent(NetState.NET_NULL))
            }
        }
    }

//    interface INetworkStateChangeCallback {
//        /**
//         * 判断网络是否可用
//         *
//         * @param available 网络是否可用
//         */
//        fun onNetworkAvailable(available: Boolean)
//    }
}
